File: TASC-Intern-Batch9/pharmacy/src/main/java/com/project/pharmacy/service/BlogService.java
package com.project.pharmacy.service;

import com.project.pharmacy.dto.request.BlogRequest;
import com.project.pharmacy.dto.response.ApiResponse;
import com.project.pharmacy.dto.response.BlogResponse;
import com.project.pharmacy.dto.response.PageResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface BlogService {
    ApiResponse<PageResponse<List<BlogResponse>>> getAllBlogs(int pageIndex, int pageSize, String title, String category);
    ApiResponse<BlogResponse> getBlogBySlug(String slug);
    ApiResponse<BlogResponse> getBlogById(Long id);
    ApiResponse<BlogResponse> createBlog(BlogRequest request, MultipartFile thumbnail);
    ApiResponse<BlogResponse> updateBlog(Long id, BlogRequest request, MultipartFile thumbnail);
    ApiResponse<Void> deleteBlog(Long id);
}

File: pharmacy-v1/fe/fe-client/src/api/blogService.ts
import type { ApiResponse, PageResponse } from '../types';
import type { Blog } from '../types/blog.types';
import axiosClient from './axiosClient';
const blogService = {
  getAllBlogs: (pageIndex: number = 1, pageSize: number = 10, title?: string, categorySlug?: string) => {
    const params: any = { pageIndex, pageSize };
    if (title) params.title = title;
    if (categorySlug) params.category = categorySlug; 

    return axiosClient.get<ApiResponse<PageResponse<Blog[]>>>('/blogs', { params });
  },

  getBlogBySlug: (slug: string) => {
    return axiosClient.get<ApiResponse<Blog>>(`/blogs/slug/${slug}`);
  }
};

export default blogService;